### DISCUSSION TITLE
Regarding error with spektral.chem module

### DISCUSSION THREAD

**Discussion by gaurav-tikhe** [danielegrattarola/spektral#219] [Q&A]
Hello Sir/Madam,
I am student of mathematical modelling and simulation and using the Spektral library for my project work. 
Based on the link below, I am trying to run the same code but while importing this "from spektral.chem import sdf_to_nx" I am getting an error saying No module named spektral.chem.

[https://www.discngine.com/blog/2020/2/26/molecular-graphs-as-input-for-neural-networks](url)

Can you please help me regarding this issue

**Comment by danielegrattarola** [comment]
Hi, 

the `chem` module is not part of Spektral anymore, and some of the functions used in that article are buggy/not fully tested.
However, the code in that example is very similar to this: https://github.com/danielegrattarola/spektral/blob/master/examples/graph_prediction/qm9_ecc.py

Try and see if you can adapt this example to your use case.
Cheers

**Reply by matine-g** [reply]
hello, I need to run a code which use chem module in spektral package. would you please tell me how can I access to that module or if its not available how can i get get_atomic_number, and get_atomic_symbol?
 thank you

**Reply by danielegrattarola** [reply]
@matine-g You can try browsing the code at a previous date like here:

https://github.com/danielegrattarola/spektral/blob/2b131e04e74b46cb3294af3bf4b475fcefacea84/spektral/chem.py

There could be more recent versions of that file, I haven't checked in detail when I removed it. Also, as I was saying in the top comment, that code is untested and potentially buggy.

**Comment by ErezKaminski** [comment]
I ran across this issue as well. What is missing in the example you linked @danielegrattarola is the ability to load my own SDF files. Is there a way one can do that in Spektral?

**Comment by danielegrattarola** [comment]
There is `spektral.utils.io` that has a `load_sdf` function to read an SDF file, but it is undocumented and meant for internal use.
You can look at `spektral.datasets.qm9` for a template class that you can adapt to read your own SDF files

Cheers
